tool/teleport/common/teleport.go

```diff
@@ -327,26 +345,54 @@ func (flags *dumpFlags) CheckAndSetDefaults() error {
 	if flags.testConfigFile != "" && flags.output != teleport.SchemeStdout {
 		return trace.BadParameter("only --output or --test can be set, not both")
 	}
-	if flags.output == "" || flags.output == teleport.SchemeFile {
-		flags.output = teleport.SchemeFile + "://" + defaults.ConfigFilePath
-	} else if flags.output == teleport.SchemeStdout {
-		flags.output = teleport.SchemeStdout + "://"
+
+	err := checkConfigurationFileVersion(flags.Version)
+	if err != nil {
+		return trace.Wrap(err)
 	}
 
+	flags.output = checkAndReturnOutputDefault(flags.output)
+	return nil
+}
+
+type createDatabaseConfigFlags struct {
+	config.DatabaseSampleFlags
+	// output is the destination to write the configuration to.
+	output string
+}
+
+func (flags *createDatabaseConfigFlags) CheckAndSetDefaults() error {
+	flags.output = checkAndReturnOutputDefault(flags.output)
+	return nil
+}
+
+func checkAndReturnOutputDefault(output string) string {
```

Comment:
Nit: Something like `normalizeOutput` would probably be a bit more suitable method name.